addons/binding/org.openhab.binding.lametrictime/ESH-INF/thing/weather-app.xml

```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="lametrictime"
+                          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                          xmlns:thing="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0"
+                          xsi:schemaLocation="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0 http://eclipse.org/smarthome/schemas/thing-description-1.0.0.xsd">
+    <thing-type id="weatherApp" listed="true">
+        <supported-bridge-type-refs>
+            <bridge-type-ref id="device" />
+        </supported-bridge-type-refs>
+        
+        <label>LaMetric Time Weather App</label>
+        <description>Provides control for the weather app on the LaMetric Time</description>
+        
+        <channels>
+            <channel id="forecast" typeId="triggerActionType">
+                <label>Forecast</label>
+                <description>Show the forecast</description>
```

Comment:
The device shows the current conditions all the time when you have the weather app activated. It shows the forecast momentarily when you send a request for it. Therefore, the on/off pattern doesn't really work. You could add a rule to automatically turn off the forecast switch as soon as you press it so that it acts more like a momentary switch than a toggle.

The main problem with a device like LaMetric Time and OH is that OH expects channels to have state and the LaMetric Time doesn't provide a lot of status information. For example, it doesn't say which app is active or what the clock alarm is set to or which station is playing on the radio. Most of the interaction is one-way towards the device. It would be nice if we had something like trigger channels, but in reverse. Something that simply sends information to a thing and expects nothing in return and no state information to be provided or retained in OH.

Reply:
Yes, so here I would suggest a similar solution as mentioned above: Instead of a Switch, use a String channel and define "FORECAST" as the only option for it. Once we have https://github.com/eclipse/smarthome/issues/4942 in place, this will nicely render a single button for this (and not showing a toggle switch). One day we might also have a solution for https://github.com/eclipse/smarthome/issues/595, so that there won't be any state changes for that channel, so that it effectively becomes a "stateless command only channel".

Reply:
I'll change this and the "stopAlarm" channel as well.

Reply:
I found an issue with the command channels. If there is only one option (e.g. weather forecast), once you set it, you can't use it again because there is no way to change the state. To avoid this, I'm clearing the state in the handler using `updateState(channelUID, new StringType())`. This works well for the Basic UI, but the Paper UI doesn't get an update to clear out the state value. If you refresh the Paper UI, it does have the updated empty value.

Do you consider this a bug in the Paper UI or is it wrong for me to change the state while handling the command?

Reply:
Both :-)
The Paper UI should react on such state changes, so I'd consider it a bug (@htreu, would you agree?) in the Paper UI.
Doing such a state change shouldn't be necessary, but the issue is that there is currently no way to suppress the auto-update (see https://github.com/eclipse/smarthome/issues/595), thus you cannot prevent the stateless "command channel" to suddenly have a state.
This will hopefully be addressed soon through https://github.com/eclipse/smarthome/pull/5011.